src/otg/datasource/gnomad/ld.py

```diff
@@ -313,6 +315,103 @@ def as_ld_index(
             _schema=LDIndex.get_schema(),
         )
 
+    @staticmethod
+    def _get_value_from_row(row: Row | None, column: str) -> int | None:
+        """Extract index value from a row if exists.
+
+        Args:
+            row (Row | None): One row from a dataframe
+            column (str): column label we want to extract.
+
+        Returns:
+            int | None: _description_
+        """
+        if isinstance(row, Row) and column in row:
+            return row[column]
+        else:
+            return None
+
+    def get_ld_variants(
+        self: GnomADLDMatrix,
+        gnomad_ancestry: str,
+        chromosome: str,
+        start: int,
+        end: int,
+    ) -> DataFrame:
+        """Return melted LD table with resolved variant id based on ancestry and genomic location.
+
+        Args:
+            gnomad_ancestry (str): GnomAD major ancestry label eg. `nfe`
+            chromosome (str): chromosome label
+            start (int): window upper bound
+            end (int): window lower bound
+
+        Returns:
+            DataFrame: square LD matrix resolved to variants.
+        """
+        # Extracting locus:
+        ld_index_df = (
+            self._process_variant_indices(
+                hl.read_table(self.ld_index_raw_template.format(POP=gnomad_ancestry)),
+                self.grch37_to_grch38_chain_path,
+            )
+            .filter(
+                (f.col("chromosome") == chromosome)
+                & (f.col("position") >= start)
+                & (f.col("position") <= end)
+            )
+            .select("chromosome", "position", "variantId", "idx")
+            .persist()
+        )
+        # +----------+---------+--------------------+--------+
+        # |chromosome| position|           variantId|     idx|
+        # +----------+---------+--------------------+--------+
+        start_index = self._get_value_from_row(
+            ld_index_df.orderBy(f.col("position").asc()).first(), "idx"
```

Comment:
Perhaps use window? https://github.com/opentargets/genetics_etl_python/blob/1b9a37b88cedaee2e62e02900d463292f794d283/src/otg/common/spark_helpers.py#L157

Reply:
I don't think it helps: I don't need to window, I need the minimum from the entire dataframe and also when I get the value from the returned dataframe I might as well fall back to `None`.